geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -111,24 +103,20 @@ class HBaseIndexAdapter(ds: HBaseDataStore) extends IndexAdapter[HBaseDataStore]
               case NonFatal(e) => logger.warn("Error checking dynamic jar path:", e); None
             }
           }
-
-          def addCoprocessor(clazz: Class[_ <: Coprocessor], desc: HTableDescriptor): Unit = {
-            val name = clazz.getCanonicalName
-            if (!desc.getCoprocessors.contains(name)) {
-              logger.debug(s"Using coprocessor path ${coprocessorUrl.orNull}")
-              // TODO: Warn if the path given is different from paths registered in other coprocessors
-              // if so, other tables would need updating
-              HBaseVersions.addCoprocessor(desc, name, coprocessorUrl)
-            }
-          }
-
           // if the coprocessors are installed site-wide don't register them in the table descriptor
           val installed = Option(conf.get(CoprocessorHost.USER_REGION_COPROCESSOR_CONF_KEY))
           val names = installed.map(_.split(":").toSet).getOrElse(Set.empty[String])
-          AllCoprocessors.foreach(c => if (!names.contains(c.getCanonicalName)) { addCoprocessor(c, descriptor) })
+          if (names.contains(CoprocessorClass)) { None } else {
+            logger.debug(s"Using coprocessor path ${coprocessorUrl.orNull}")
+            // TODO: Warn if the path given is different from paths registered in other coprocessors
+            // if so, other tables would need updating
+            Some(CoprocessorClass -> coprocessorUrl)
+          }
         }
 
-        try { admin.createTableAsync(descriptor, splits.toArray) } catch {
+        try {
+          HBaseVersions.createTableAsync(admin, name, cols, bloom, compression, encoding, None, coprocessor, splits)
```

Comment:
Alright, good, we do make an async call...

Does this call make sense to be async?  Or should it block?

Reply:
well we actually block a few lines down